dgraph/cmd/bulk/reduce.go

```diff
@@ -127,11 +136,23 @@ func (r *reducer) createBadger(i int) *badger.DB {
 
 	// Zero out the key from memory.
 	opt.EncryptionKey = nil
+	return db
+}
 
+func (r *reducer) createBadger(i int) *badger.DB {
+	db := r.createBadgerInternal(r.opt.shardOutputDirs[i])
 	r.dbs = append(r.dbs, db)
 	return db
 }
 
+func (r *reducer) createTmpBadger() *badger.DB {
+	tmpDir, err := ioutil.TempDir(r.opt.TmpDir, "split")
+	x.Check(err)
+	db := r.createBadgerInternal(tmpDir)
```

Comment:
createbadgerInteral creates a badger instance with compression enabled. I don't think we need compression for temporary DB.